### PR TITLE
Update pre-commit config and black

### PR DESCRIPTION
**Pull request by mattwthompson**
I recently scaled back pre-commit's autoupdater to quarterly since _most_ patch changes of linters/autoformatters are inconsequential for our codebase. This is not one such case; black changed its opinion on how the `**` power operator should look, resulting in a relatively large number of code changes. They [pinky promise](https://black.readthedocs.io/en/stable/the_black_code_style/index.html#stability-policy) changes will only happen once a year now that it's formally out of beta, so this is presumably our once annual `black openff` call.

### CODE REVIEW

File: openff/toolkit/topology/molecule.py [hunk 1]
```diff
@@ -4586,7 +4586,8 @@ def from_iupac(
             molecule = toolkit.from_iupac(
                 iupac_name,
                 allow_undefined_stereo=allow_undefined_stereo,
-                _cls=cls ** kwargs,
+                _cls=cls,
+                **kwargs,
```
Comment (mattwthompson) [inline openff/toolkit/topology/molecule.py:4590]: This ... probably wasn't working before.

Comment (mattwthompson) [reply openff/toolkit/topology/molecule.py:4590]: This is the only manual change I made (adding a comma).

Comment (j-wags) [reply openff/toolkit/topology/molecule.py:4590]: (not blocking) Oy vey. I hate this "registry or wrapper" pattern, so many untested cases! Let's never use this design pattern ever again.

File: openff/toolkit/utils/serialization.py [hunk 2]
```diff
@@ -318,7 +318,7 @@ def from_yaml(cls, serialized):
         yaml.SafeDumper.add_representer(
             OrderedDict,
             lambda dumper, value: self._represent_odict(
-                dumper, u"tag:yaml.org,2002:map", value
+                dumper, "tag:yaml.org,2002:map", value
```
Comment (j-wags) [inline openff/toolkit/utils/serialization.py:321]: (not blocking) this diff looks scary but it just seems to be cosmetic https://stackoverflow.com/questions/2464959/whats-the-u-prefix-in-a-python-string

**Review by j-wags — APPROVED** [review]
LGTM! Thanks @mattwthompson

File: openff/toolkit/utils/serialization.py (hunk 2 above)
Comment (mattwthompson) [reply openff/toolkit/utils/serialization.py:321]: I was a little spooked by this as well

**Pull request merged**